src/pairs.rs

```diff
@@ -183,15 +189,17 @@ where
         ..shape
     };
     let lhs_result = lhs
-        .rewrite(context, lhs_shape)
+        .rewrite_result(context, lhs_shape)
         .map(|lhs_str| format!("{}{}", pp.prefix, lhs_str))?;
 
     // Try to put both lhs and rhs on the same line.
     let rhs_orig_result = shape
         .offset_left(last_line_width(&lhs_result) + pp.infix.len())
         .and_then(|s| s.sub_width(pp.suffix.len()))
-        .and_then(|rhs_shape| rhs.rewrite(context, rhs_shape));
-    if let Some(ref rhs_result) = rhs_orig_result {
+        .max_width_error(shape.width, rhs.span())
+        .and_then(|rhs_shape| rhs.rewrite_result(context, rhs_shape));
```

Comment:
In `rewrite_pair`, I added the bound for the `Spanned` trait for both lhs and rhs. However, this reports the max width error with only `rhs.span` since we first rewrite lhs and then compute the shape for rhs. Do you think it would be better to set the span as `lhs.span` to `rhs.span` or just `rhs.span`?

Reply:
I think it's fine to just use the `rhs` span in this case, since that's what we're failing to find space for, though I could also see us using the complete span from both the `lhs` and the `rhs`. Should be able to use [`Span::to`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/struct.Span.html#method.to) to easily get that complete span.